solidity/contracts/libraries/staking/TopUps.sol

```diff
@@ -73,4 +136,15 @@ library TopUps {
         delete self.topUps[operator];
         return topUp.amount;
     }
+
+    /// @notice Returns true if the given operatorParams indicate that the
+    /// operator is undelegating its stake or that it completed stake
+    /// undelegation.
+    /// @param operatorParams Parameters of the operator, as stored in the
+    /// staking contract.
+    function isUndelegating(uint256 operatorParams)
+        internal view returns (bool) {
```

Comment:
Can this function be updated to `pure`?
```suggestion
        internal pure returns (bool) {
```
If so, can we do the same in the `TokenStaking`'s one?

Reply:
Nope, it's not going to work:
```
TopUps.sol:148:41: TypeError: Function declared as pure, but this expression (potentially) reads from the environment or state and thus requires "view".
        return (undelegatedAt != 0) && (block.timestamp > undelegatedAt);
```